truck_tour_exe.py

```diff
@@ -0,0 +1,81 @@
+# Hacker Rank Problem Link: https://www.hackerrank.com/challenges/truck-tour/problem
+
+#!/bin/python3
+
+import os
+import sys
+
+#
+# Complete the truckTour function below.
+#
+def truckTour(petrolpumps, N):
+    #
+    # Write your code here.
+    #
+    pump = []
+    dist = []
+    start=0
+    curr=0
+    total=0
+    for pair in petrolpumps:
+        pump.append(pair[0])
+        dist.append(pair[1])
+    for i in range(N):
+        total = total + pump[i] - dist[i]
+        curr = curr + pump[i] - dist[i]
+        if curr < 0:
+            curr = 0
+            start = i + 1

```

Comment:
I don't think this working. It may work on the minimal input from the exercise but we should be able to construct a case where this wouldn't return the right answer. In particular, what I am concerned about is not actually testing the full cycle if you restart. While |total| > 0 ensures that there is a solution, it doesn't ensure that the one that can go to N is the one.